calico_containers/integrations/kubernetes/calico_kubernetes.py

```diff
@@ -95,19 +95,24 @@ def _configure_interface(self):
         return ep
 
     def _get_node_ip(self):
-        """Determine the IP for the host node.
-
-        This hits a well-known IP (the Google public DNS).
+        """ 
+        Determine the IP for the host node.
         """
-        # We'd like to use the k8s API here, but it doesn't work in the Vagrant
-        # case since the node's name is set to its IP (so we have no way of
-        # getting from hostname=>IP).
-        # TODO: do this more reliably by parsing 'ip
-        s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-        s.connect(('8.8.8.8', 80))
-        ip = s.getsockname()[0]
-        s.close()
-        return ip
+        # Compile list of addresses on network, return the first entry.
+        # Try IPv4 and IPv6.
+        addrs = get_host_ips(version=4) or get_host_ips(version=6)
+
+        try:
+            addr = addrs[0]
+            print('Using IP Address %s' % (addr))
+            return addr
+        # If both get_host_ips return empty lists, print msg and return None
+        except IndexError:
+            print('No Valid IP Address Found for Host')
+            return None
+
+        return None
```

Comment:
This return is redundant - It won't ever be hit because of the try / except above.

Also, I think we shouldn't be returning None from this method at all.  If we can't find an IP address, we should log a clear error and sys.exit